python/hsfs/core/feature_view_engine.py

```diff
@@ -103,27 +104,48 @@ def get_batch_query(
         with_label=False,
         training_dataset_version=None,
     ):
-        return self._feature_view_api.get_batch_query(
-            feature_view_obj.name,
-            feature_view_obj.version,
-            start_time,
-            end_time,
-            training_dataset_version=training_dataset_version,
-            is_python_engine=engine.get_type() == "python",
-            with_label=with_label,
-        )
+        try:
+            return self._feature_view_api.get_batch_query(
+                feature_view_obj.name,
+                feature_view_obj.version,
+                util.convert_event_time_to_timestamp(start_time),
+                util.convert_event_time_to_timestamp(end_time),
+                training_dataset_version=training_dataset_version,
+                is_python_engine=engine.get_type() == "python",
+                with_label=with_label,
+            )
+        except exceptions.RestAPIError as e:
+            if e.response.json().get("errorCode", "") == 270172:
+                raise ValueError(
+                    "Cannot generate dataset(s) from the given start/end time because"
+                    " event time column is not available in one or more feature groups."
```

Comment:
It is required to have at least 1 FG with event time. I suggest to be like `non of the feature groups contains event time column`

Reply:
In the backend we actually check the left feature group only. I updated the message.